### ISSUE TITLE
Bug in authorizeFileSystemUse

### ISSUE THREAD

**Issue by STAT-UP** [RevolutionAnalytics/checkpoint#242]
Hi,

the current version of `authorizeFileSystemUse` throws an infinite recursion error on my system.

The problem is `interactive = interactive()` in the parameter definition:
```
authorizeFileSystemUse <- function(checkpointLocation, interactive = interactive()) 
```

To fix it, the line should read (following foreach's parameter naming convention):
```
authorizeFileSystemUse <- function(.checkpointLocation, .interactive = interactive()) 
```

or (not requiring further code changes):
```
authorizeFileSystemUse <- function(checkpointLocation, interactive = base::interactive()) 
```

Thanks,
Stefan

**Comment by sven-stodtmann** [comment]
Hi,
it seems that the current version of authorizeFileSystemUse returns nothing, as the last line is
```invisible()```
this can lead to 
```
Error in if (!file.exists(x)) dir.create(x, recursive = TRUE, showWarnings = FALSE) : 
  argument is of length zero
```
I guess what was intended here is 
```invisible(checkpointRoot)```

**Comment by gwaygenomics** [comment]
@andrie any updates on this? I am getting similar error messages where it worked great before

**Comment by andrie** [comment]
@sven-stodtmann The function has always returned NULL if successful.

Can you please provide a reproducible example where this causes a problem?

**Comment by sven-stodtmann** [comment]
@andrie my bad, it seems this was caused by the system I tried to run it on due to a non set R_LIBS_USER environment variable which I passed as the checkpoint location.

**Comment by harrismcgehee** [comment]
@andrie, the fix #242 fixed the error I was experiencing. Thanks!

**Comment by gwaygenomics** [comment]
can also confirm that 08a1382b7dc4105c8c77f74abe245b7152634a9d fixed the issue. Thanks! Looking forward to this being merged into master
